src/main/java/org/jfrog/hudson/pipeline/declarative/utils/DeclarativePipelineUtils.java

```diff
@@ -0,0 +1,171 @@
+     * @return Artifactory server.
+     */
+    public static ArtifactoryServer getArtifactoryServer(TaskListener listener, Run build, FilePath ws, StepContext context, String serverId) throws IOException, InterruptedException {
+        String buildNumber = BuildUniqueIdentifierHelper.getBuildNumber(build);
+        BuildDataFile buildDataFile = DeclarativePipelineUtils.readBuildDataFile(listener, ws, buildNumber, CreateServerStep.STEP_NAME, serverId);
+        if (buildDataFile == null) {
+            GetArtifactoryServerExecutor getArtifactoryServerExecutor = new GetArtifactoryServerExecutor(build, context, serverId);
+            getArtifactoryServerExecutor.execute();
+            return getArtifactoryServerExecutor.getArtifactoryServer();
+        }
+        JsonNode jsonNode = buildDataFile.get(CreateServerStep.STEP_NAME);
+        ArtifactoryServer server = Utils.mapper().treeToValue(jsonNode, ArtifactoryServer.class);
+        String credentialsId = jsonNode.get("credentialsId").asText();
+        if (!credentialsId.isEmpty()) {
+            server.setCredentialsId(credentialsId);
+        } else {
+            server.setUsername(jsonNode.get("username").asText());
+            server.setPassword(jsonNode.get("password").asText());
+        }
+        return server;
+    }
+
+    /**
+     * Create build info id: <buildname>_<buildnumber>.
+     * @param build - Step's build.
+     * @param customBuildName - Step's custom build name if exist.
+     * @param customBuildNumber - Step's custom build number if exist.
+     * @return build info id: <buildname>_<buildnumber>.
+     */
+    public static String createBuildInfoId(Run build, String customBuildName, String customBuildNumber) {
+        return StringUtils.defaultIfEmpty(customBuildName, BuildUniqueIdentifierHelper.getBuildName(build)) + "_" +
+                StringUtils.defaultIfEmpty(customBuildNumber, BuildUniqueIdentifierHelper.getBuildNumber(build));
+    }
+
+    /**
+     * Get build info as defined in previous rtBuildInfo{...} scope.
+     * @param listener - Step's listener.
+     * @param ws - Step's workspace.
+     * @param build - Step's build.
+     * @param customBuildName - Step's custom build name if exist.
+     * @param customBuildNumber - Step's custom build number if exist.
+     * @return build info object as defined in previous rtBuildInfo{...} scope or a new build info.
+     */
+    public static BuildInfo getBuildInfo(TaskListener listener, FilePath ws, Run build, String customBuildName, String customBuildNumber) throws IOException, InterruptedException {
+        String jobBuildNumber = BuildUniqueIdentifierHelper.getBuildNumber(build);
+        String buildInfoId = createBuildInfoId(build, customBuildName, customBuildNumber);
+
+        BuildDataFile buildDataFile = DeclarativePipelineUtils.readBuildDataFile(listener, ws, jobBuildNumber, BuildInfoStep.STEP_NAME, buildInfoId);
```

Comment:
**DeclarativePipelineUtils.readBuildDataFile** can be replaced with **readBuildDataFile**.